Use privateassets=all for analyzer packages to avoid propagating analyzer references

**Describe the bug**

Referencing the Bicep code (say Bicep.Core as an example) from another codebase will cause you to be inundated with build warnings.

Example (from the 1030 warnings I get 😆):

```txt
/Users/ryan/github.com/rynowak/tycep/submodules/tye/src/Microsoft.Tye.Core/KubernetesManifestGenerator.cs(500,45): warning CA1305: The behavior of 'int.ToString()' could vary based on the current user's locale settings. Replace this call in 'KubernetesManifestGenerator.AddProbe(ServiceBuilder, YamlMappingNode, ProbeBuilder, string)' with a call to 'int.ToString(IFormatProvider)'. [/Users/ryan/github.com/rynowak/tycep/submodules/tye/src/Microsoft.Tye.Core/Microsoft.Tye.Core.csproj]
```

**To Reproduce**

Build a project that uses code like `int x = 17; Console.WriteLine(x.ToString())`. 

Add the bicep repo as a submodule and reference `Bicep.Core.csproj`.

**Additional context**

The analyzer packages I'm aware of are referenced here: https://github.com/Azure/bicep/blob/main/src/Directory.Build.props#L17

Since these don't tag the references as private they *follow* project and package references, so that anyone referencing bicep picks them up too.

--- a/src/Directory.Build.props
+++ b/src/Directory.Build.props
@@ -14,8 +14,12 @@
       <Version>3.3.37</Version>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.1"/>
-    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118"/>
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.1">
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
   <ItemGroup>

